### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/DisplayConstants.java [hunk 1]
```diff
@@ -319,6 +319,9 @@
 	public static final String ENTITY_DESCRIPTION_SHOW_TIPS_TEXT = "Formatting Guide";
 	public static final String MARKDOWN_CSS_CLASSNAME = "from-markdown";
 	public static final String ENTITY_DESCRIPTION_ATTACHMENT_PREFIX = "Attachment/entity/";
+	public static final String ENTITY_STORAGE_NOT_FOUND_ERROR = "Unable to determine the usage associated with entity id: ";
+	public static final String LOCATIONABLE_NO_FILE_FOUND = "No File Found";
+	public static final String LOCATIONABLE_NO_FILE_FOUND_DETAIL = "A file must be uploaded (or externally linked to) before analysis can be run.";
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/DisplayConstants.java]: Dave: change to "It is recommended that you upload a file or link to an external location as this object is intended to reference a file."

File: src/main/java/org/sagebionetworks/web/client/DisplayConstants.java [hunk 2]
```diff
@@ -214,7 +214,7 @@
 	public static final String LABEL_SEARCH = "Search";
 	public static final String LABEL_PROJECT_NAME = "Project Name";
 	public static final String LABEL_PROJECT_CREATED = "Project Created";
-	public static final String TEXT_UPLOAD_FILE = "Upload File";
+	public static final String TEXT_UPLOAD_FILE = "Upload";
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/DisplayConstants.java]: Dave: change to "Upload or Link to File"

File: src/main/java/org/sagebionetworks/web/client/IconsImageBundle.java [hunk 3]
```diff
@@ -176,12 +176,18 @@
 	@Source("images/icons/navigation-090-button-16.png")
 	ImageResource NavigateUp16();
 
+	@Source("images/icons/navigation-090-button-16-blue.png")
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/IconsImageBundle.java]: remove blue

File: src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+	public static String getLocationablePath(EntityBundle bundle) {
+		String locationPath = null;
+		if (!(bundle.getEntity() instanceof Locationable))
+			throw new IllegalArgumentException("Bundle must reference a Locationable entity");
+		if (((Locationable)bundle.getEntity()).getLocations() != null && ((Locationable)bundle.getEntity()).getLocations().size() > 0)
+			locationPath = ((Locationable)bundle.getEntity()).getLocations().get(0).getPath();
+		return locationPath;
+	}
+	
+	public static boolean isDataPossiblyWithinLocationable(EntityBundle bundle, boolean isLoggedIn) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java]: move to util or presenter, and add test

File: src/main/java/org/sagebionetworks/web/client/widget/entity/EntityMetadataViewImpl.java [hunk 5]
```diff
@@ -246,10 +257,25 @@ public void setEntityBundle(EntityBundle bundle, boolean readOnly) {
...
 		AbstractImagePrototype synapseIconForEntity = AbstractImagePrototype.create(DisplayUtils.getSynapseIconForEntity(e, DisplayUtils.IconSize.PX24, icons));
 		synapseIconForEntity.applyTo(entityIcon);
-
+		
 		setEntityName(e.getName());
 		setEntityId(e.getId());
-
+		boolean isLocationable = (e instanceof Locationable);
+		//show the entity name if this isn't locationable, or if it has no data.
+		boolean isEntityNamePanelVisible = !isLocationable || !LocationableTitleBarViewImpl.isDataPossiblyWithinLocationable(bundle, !presenter.isAnonymous());
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/EntityMetadataViewImpl.java]: change to &&

File: src/main/java/org/sagebionetworks/web/client/widget/entity/download/LocationableUploader.java [hunk 6]
```diff
@@ -71,6 +71,7 @@ public LocationableUploader(
 	}		
 		
 	public Widget asWidget(EntityBundle entityBundle) {
+		this.view.setPresenter(this);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/download/LocationableUploader.java:74]: should not be singleton

File: src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+		else if (bundle.getUnmetAccessRequirements() != null && bundle.getUnmetAccessRequirements().size() > 0)
+			hasData = true;
+		//else, if it has a locations list whose size is > 0
+		else if (((Locationable)bundle.getEntity()).getLocations() != null && ((Locationable)bundle.getEntity()).getLocations().size() > 0)
+			hasData = true;
+		return hasData;
+	}
+	
+	@Override
+	public void createMenu(
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java]: rename to createTitleBar

File: src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+			boolean readOnly) {
+		
+		Entity entity = entityBundle.getEntity();
+
+		UserSessionData sessionData = authenticationController.getLoggedInUser();
+		UserProfile userProfile = (sessionData==null ? null : sessionData.getProfile());
+
+		if(downloadButton == null){
+			downloadButton = licensedDownloader.asWidget(entityBundle, userProfile);
+			DisplayUtils.addTooltip(this.synapseJSNIUtils, downloadButton, DisplayConstants.BUTTON_DOWNLOAD, TOOLTIP_POSITION.BOTTOM);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java:106]: move download button to the right of entity name display and reinstate entity icon (eventually to be a type icon)

File: src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+		}
+		
+		if (entity instanceof Locationable) {
+			
+			//configure this view based on if this entity has locations to download
+			Locationable locationable = (Locationable)entity;
+			
+			if (isDataPossiblyWithinLocationable(entityBundle, authenticationController.isLoggedIn())) {
+				HorizontalPanel panel = new HorizontalPanel();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java:115]: redesign layout so long entity/file name, synapse id and file size/md5 info display inline

File: src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+				panel.setVerticalAlign(VerticalAlignment.MIDDLE);
+
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/file/LocationableTitleBarViewImpl.java:117]: break long run of commands into smaller private methods

File: src/main/webapp/Portal.css [hunk 11]
```diff
@@ -474,6 +475,10 @@ blockquote.from-markdown {
     padding: 20px;
 }
 
+.file-md5 { font-style: italic; }
```
Comment (jay-hodgson) [inline src/main/webapp/Portal.css]: change to font-italic

File: src/main/webapp/Portal.css [hunk 12]
```diff
@@ -474,6 +475,10 @@ blockquote.from-markdown {
     padding: 20px;
 }
 
+.file-md5 { font-style: italic; }
+
+.file-size {}
```
Comment (jay-hodgson) [inline src/main/webapp/Portal.css]: x

**Comment by dburdick** [comment]
looks good!

**Pull request merged**